framework/PostProcessors/ETimporter.py

```diff
@@ -0,0 +1,262 @@
+    """
+      Function to read the portion of the xml input that belongs to this specialized class
+      and initialize some stuff based on the inputs got
+      @ In, xmlNode, xml.etree.Element, Xml element node
+      @ Out, None
+    """
+    for child in xmlNode:
+      if child.tag == 'fileFormat':
+        if child.text not in self.allowedFormats:
+          self.raiseAnError(IOError, 'ETimporterPostProcessor Post-Processor ' + self.name + ', format ' + child.text + ' : is not supported')
+        else:
+          self.ETformat = child.text
+      else:
+        self.raiseAnError(IOError, 'ETimporterPostProcessor Post-Processor ' + self.name + ', node ' + child.tag + ' : is not recognized')
+
+  def run(self, input):
+    """
+      This method executes the postprocessor action.
+      @ In,  input, object, object containing the data to process. (inputToInternal output)
+      @ Out, None
+    """
+    if self.ETformat == 'OpenPSA':
+      self.outputDict = self.runOpenPSA(input)
+
+  def runOpenPSA(self, input):
+    """
+      This method executes the postprocessor action.
+      @ In,  input, object, object containing the data to process. (inputToInternal output)
+      @ Out, None
+    """
+    fileToOpen = input[0].getPath() + input[0].getFilename()
+    root = ET.parse(fileToOpen)
+    eventTree = root.findall('initial-state')
+
+    if len(eventTree) > 1:
+      self.raiseAnError(IOError,'ETimporter: more than one initial-state identified')
+
+    eventTree = eventTree[0]
+
+    ## These outcomes will be encoded as integers starting at 0
+    outcomes = []
+
+    ## These self.variables will be mapped into an array where there index
+    self.variables = []
+    values = {}
+
+    for node in root.findall('define-functional-event'):
+      event = node.get('name')
+
+      ## First, map the variable to an index by placing it in a list
+      self.variables.append(event)
+
+      ## Also, initialize the dictionary of values for this variable so we can
+      ## encode them as integers as well
+      values[event] = []
+
+      ## Iterate through the forks that use this event and gather all of the
+      ## possible states
+      for fork in self.findAllRecursive(eventTree, 'fork'):
+        if fork.get('functional-event') == event:
+          for path in fork.findall('path'):
+            state = path.get('state')
+            if state not in values[event]:
+              values[event].append(state)
+
+    ## Iterate through the sequences and gather all of the possible outcomes
+    ## so we can numerically encode them latter
+    for node in root.findall('define-sequence'):
+      outcome = node.get('name')
+      if outcome not in outcomes:
+        outcomes.append(outcome)
+
+    self.raiseADebug("ETimporter variables identified: " + str(format(self.variables)))
+
+    d = len(self.variables)
+    n = len(self.findAllRecursive(eventTree, 'sequence'))
+
+    self.pointSet = -1 * np.ones((n, d + 1))
```

Comment:
Same as before, do not use the state variables if they will change during the run time.